packages/office-addin-dev-settings/test/files/manifests/manifest.json

```diff
@@ -1,121 +1,147 @@
+                "pinnable": false
+                }
             ]
           }
-        ]
-      }
-    ]
+        },
+        {
+          "id": "ExecuteFunction",
+          "type": "general",
+          "code": {
+            "page": "https://localhost:3000/commands.html",
+            "script": "https://localhost:3000/commands.js"
+          },
+          "lifetime": "short",
+          "actions": {
+              "items": [
+                {
+                  "id": "ExecuteFunction.action",
+                  "type": "execution",
+                  "name": "action"
+                }
+              ]
+          }
+        }
+      ],
+      "ribbons": [
+        {
+          "contexts": [ "readMail" ],
+          "tabs": [
+            {
+              "id": "TabDefault",
+              "groups": [
+                {
+                  "id": "msgReadGroup",
+                  "label": "Contoso Add-in",
+                  "icons": [
+                    { "size": 16, "file": "https://localhost:3000/assets/icon-16.png" },
+                    { "size": 32, "file": "https://localhost:3000/assets/icon-32.png" },
+                    { "size": 80, "file": "https://localhost:3000/assets/icon-80.png" }
+                  ],
+                  "controls": [
+                    {
+                      "id": "msgReadOpenPaneButton",
+                      "type": "button",
+                      "label": "Show Taskpane",
+                      "icons": [
+                        { "size": 16, "file": "https://localhost:3000/assets/icon-16.png" },
+                        { "size": 32, "file": "https://localhost:3000/assets/icon-32.png" },
+                        { "size": 80, "file": "https://localhost:3000/assets/icon-80.png" }
+                      ],
+                      "supertip": {
+                        "title": "Show Taskpane",
+                        "description": "Opens a pane displaying all available properties."
+                      },
+                      "actionId": "ShowTaskpane.show"
+                    },
+                    {
+                      "id": "ActionButton",
+                      "type": "button",
+                      "label": "Perform an action",
+                      "icons": [
+                        { "size": 16, "file": "https://localhost:3000/assets/icon-16.png" },
+                        { "size": 32, "file": "https://localhost:3000/assets/icon-32.png" },
+                        { "size": 80, "file": "https://localhost:3000/assets/icon-80.png" }
+                      ],
+                      "supertip": {
+                        "title": "Perform an action",
+                        "description": "Perform an action when clicked."
+                      },
+                      "actionId": "ExecuteFunction.action"
+                    }
+                  ]
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
   }
-}
+  
```

Comment:
NIT: endl here

Reply:
??

Reply:
??

Reply:
end line* here
I meant to say to add a enter/break/end line in the end of the file